### PR TITLE
Update client-sdk-api.md

### PR DESCRIPTION
**Pull request by llbartekll**

### CODE REVIEW

File: docs/specs/clients/notify/client-sdk-api.md [hunk 1]
```diff
@@ -66,6 +66,9 @@ abstract class Client {
     onSign: (message: string) => Promise<Cacao.Signature>
   }): Promise<string>;
 
+  // deletes all subscriptions, decouples client from the registered account and pauses watchSubscription requests
```
Comment (chris13524) [inline docs/specs/clients/notify/client-sdk-api.md]: I don't think deleting all subscriptions makes sense to do here: https://walletconnect.slack.com/archives/C044SKFKELR/p1695827130071699?thread_ts=1695826302.572449&cid=C044SKFKELR

**Review by chris13524 — CHANGES_REQUESTED** [review]

File: docs/specs/clients/notify/client-sdk-api.md [hunk 2]
```diff
@@ -66,6 +66,9 @@ abstract class Client {
...
+  public abstract logOut(): Promise<Void>;
```
Comment (Elyniss) [inline docs/specs/clients/notify/client-sdk-api.md]: How about we add two methods?
mute(account: string) (old logOut())
unmute(account: string)

File: docs/specs/clients/notify/client-sdk-api.md [hunk 3]
```diff
@@ -66,6 +66,9 @@ abstract class Client {
     onSign: (message: string) => Promise<Cacao.Signature>
   }): Promise<string>;
 
+  // Unsubscribes from all subscription topics related to the most recently registered account, decouples the client from the registered account, and pauses watchSubscription requests.
```
Comment (devceline) [inline docs/specs/clients/notify/client-sdk-api.md]: What does `decouples the client from the registered account` mean

Comment (llbartekll) [reply docs/specs/clients/notify/client-sdk-api.md]: for instance service that is periodically triggering watchSubscription request is no longer aware of the account
probably depends of the implementation

Comment (devceline) [reply docs/specs/clients/notify/client-sdk-api.md]: Shouldn't it also delete the actual; identity key

Comment (Elyniss) [reply docs/specs/clients/notify/client-sdk-api.md]: The unregister is good when we want this device to forget about Notify API whatsoever. 

Clients are account agnostic. Why should we track most recently registered account? What is missing in he method is `account`. We had a phase in Chat API when we wanted to do similar thing. In the end we decided to fallback to always specifying the account for simplicity.


On the other hand,
I still think we should go mute/unmute route. Maybe not in this PR but in near future. Which would alternatively solve it.

Comment (llbartekll) [reply docs/specs/clients/notify/client-sdk-api.md]: >Shouldn't it also delete the actual; identity key

why? user may want to use this account again later

Comment (devceline) [reply docs/specs/clients/notify/client-sdk-api.md]: Then calling it `unregister` doesn't make sense

Comment (Elyniss) [reply docs/specs/clients/notify/client-sdk-api.md]: Then why would they unregister?

File: docs/specs/clients/notify/client-sdk-api.md [hunk 4]
```diff
@@ -66,6 +67,13 @@ abstract class Client {
     onSign: (message: string) => Promise<Cacao.Signature>
   }): Promise<string>;
 
+  // exposed IdentityClient.unregister method
+  // unregister identity key for given account. Removes aforementioned identity key from local storage. Unregisters device from Push Server
+  // removes any logic to perodically call `wc_notifyWatchSubscriptions`
```
Comment (chris13524) [inline docs/specs/clients/notify/client-sdk-api.md]:
```suggestion
  // "Logs out" this client from any notifications for the specified account. This involves:
  // - Stops periodically calling `wc_notifyWatchSubscriptions` and unsubscribes from all related relay topics
  // - Unregisters the client from Push Server
  // - Unregisters the Identity Key and removes it from local storage
```

**Review by chris13524 — COMMENTED** [review]
Can we cleanup the wording a bit?

**Comment by flypaper0** [comment]
Also need to include `account` param into `getActiveSubscriptions()`

**Review by devceline — APPROVED** [review]
LGTM

File: docs/specs/clients/notify/client-sdk-api.md [hunk 5]
```diff
@@ -66,6 +67,14 @@ abstract class Client {
     onSign: (message: string) => Promise<Cacao.Signature>
   }): Promise<string>;
 
+  // "Logs out" this client from any notifications for the specified account. This involves:
+  // - Stops periodically calling `wc_notifyWatchSubscriptions` and unsubscribes from all related relay topics
+  // - Unregisters the client from Push Server
+  // - Unregisters the Identity Key and removes it from local storage
```
Comment (chris13524) [inline docs/specs/clients/notify/client-sdk-api.md:73]:
```suggestion
  // - Unregisters the Identity Key and removes it from local storage
  // This is not to be confused with `deleteSubscription()` method
```

**Review by chris13524 — APPROVED** [review]

**Review by flypaper0 — APPROVED** [review]

**Review by Elyniss — APPROVED** [review]

**Pull request merged**